pkgs/applications/science/math/sage/patches/Only-test-external-software-when-all-of-sage-is.patch

```diff
@@ -0,0 +1,46 @@
+diff --git a/src/sage/doctest/control.py b/src/sage/doctest/control.py
+index 01f32fb8e4..4d83dc7b47 100644
+--- a/src/sage/doctest/control.py
++++ b/src/sage/doctest/control.py
+@@ -1253,6 +1253,7 @@ class DocTestController(SageObject):
+ 
+             self.log("Using --optional=" + self._optional_tags_string())
+             available_software._allow_external = self.options.optional is True or 'external' in self.options.optional
++            available_software._autodetect_safe = self.options.optional is True or 'sage' in self.options.optional
+             self.log("Features to be detected: " + ','.join(available_software.detectable()))
+             self.add_files()
+             self.expand_files_into_sources()
+diff --git a/src/sage/doctest/external.py b/src/sage/doctest/external.py
+index 84dae19ea5..badc61466e 100644
+--- a/src/sage/doctest/external.py
++++ b/src/sage/doctest/external.py
+@@ -409,6 +409,7 @@ class AvailableSoftware(object):
+             []
+         """
+         self._allow_external = True
++        self._autodetect_safe = True
+         # For multiprocessing of doctests, the data self._seen should be
+         # shared among subprocesses. Thus we use Array class from the
+         # multiprocessing module.
+@@ -430,6 +431,8 @@ class AvailableSoftware(object):
+             sage: 'internet' in available_software # random, optional - internet
+             True
+         """
++        if not self._autodetect_safe:
++            return False
```

Comment:
Alternatively, we could change https://github.com/sagemath/sage/blob/3da1b22c253205cc4b22772d861f2ffc3d4b2eb2/src/sage/doctest/parsing.py#L694 to check self.optional_only (which is True if and only if "sage" is not in the "--optional" command line argument). This would also work, but the doctest output would still claim features were detected and the doctest report (https://github.com/sagemath/sage/blob/3da1b22c253205cc4b22772d861f2ffc3d4b2eb2/src/sage/doctest/reporting.py#L155) would incorrectly report that tests depending on auto-detected features were run. Since the present fix only touches files in the `src/sage/doctest` directory, I think it is harmless.

(All of this assumes we'll still be running `sage -t --all --optional=sagemath_doc_html` without basic Sage tests, which is probably something upstream will break every once in a while.)